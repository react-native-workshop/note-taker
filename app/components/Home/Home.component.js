import React, {Component} from 'react';
import {View, Text, TouchableOpacity} from 'react-native';
import TextArea from '../TextArea/TextArea.component';
import PropTypes from 'prop-types';
import styles from './Home.component.style';
import Link from '../Link/Link.component';

class Home extends Component {
  addNote = () => {
    const {saveNote, title, text} = this.props;
    if (title && text) {
      saveNote({title, text});
    }
  }
  render () {
    const {setTitle, title, text, setText, linkToOpen} = this.props;
    return (
      <View style={styles.container}>
        <Text style={styles.titleHeading}> Note Title</Text>
        <TextArea style={styles.titleTextInput}
            onChangeText={setTitle} value={title} />
        <Text style={styles.textAreaTitle}> Please type your note below </Text>
        <TextArea style={styles.textArea} multiline = {true}
          onChangeText={setText} value={text}/>
        <View style={styles.bottomBarWrapper}>
          <TouchableOpacity style={styles.touchable} onPress={this.addNote}>
            <Text style={styles.saveBtn} >Save</Text>
          </TouchableOpacity>
          <Text style={styles.characterCount}>{text.length} characters</Text>
        </View>
        <Link link={linkToOpen} text='Click here to open'/>
      </View>
    );
  }
}

Home.defaultProps = {};

Home.propTypes = {
  setTitle: PropTypes.func,
  setText: PropTypes.func,
  title: PropTypes.string,
  saveNote: PropTypes.func,
  notes: PropTypes.array,
  text: PropTypes.string,
  linkToOpen: PropTypes.string
};

export default Home;
