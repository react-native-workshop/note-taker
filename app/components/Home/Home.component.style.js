import theme from '../../styles/theme.style';
import {headingText, textInput} from '../../styles/common.style';

export default {
  container: {
    flex: 1,
    paddingTop: theme.CONTAINER_PADDING,
    flexDirection: 'column',
    justifyContent: 'space-between'
  },
  titleHeading: {
    ...headingText
  },
  titleTextInput: {
    ...textInput
  },
  textAreaTitle: {
    ...headingText,
    fontWeight: theme.FONT_WEIGHT_LIGHT,
    fontStyle: 'italic'
  },
  textArea: {
    ...textInput,
    flex: 1,
    textAlignVertical: 'top',
    fontSize: theme.FONT_SIZE_MEDIUM,
    fontWeight: theme.FONT_WEIGHT_LIGHT
  },
  bottomBar: {
    flexDirection: 'row',
    alignItems: 'center'
  },
  bottomBarWrapper: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    flex: 1
  },
  saveBtn: {
    padding: 10,
    fontWeight: theme.FONT_WEIGHT_BOLD
  },
  touchable: {
    padding: 5,
  },
  characterCount: {
    padding: 10,
    fontSize: theme.FONT_SIZE_SMALL
  }
};
