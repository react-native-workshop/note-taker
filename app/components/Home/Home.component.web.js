import React, {Component} from 'react';

import TextArea from '../TextArea/TextArea.component';
import PropTypes from 'prop-types';
import styles from './Home.component.style';
import Link from '../Link/Link.component';

class Home extends Component {
  addNote = () => {
    const {saveNote, title, text} = this.props;
    if (title && text) {
      saveNote({title, text});
    }
  }
  render () {
    const {setTitle, title, text, setText, linkToOpen} = this.props;
    return (
      <div style={styles.container}>
        <p style={styles.titleHeading}> Note Title</p>
        <TextArea style={styles.titleTextInput}
            onChangeText={setTitle} value={title} />
        <p style={styles.textAreaTitle}> Please type your note below </p>
        <TextArea style={styles.textArea} multiline = {true}
          onChangeText={setText} value={text}/>
        <div style={styles.bottomBarWrapper}>
          <span style={styles.touchable} onClick={this.addNote}>
            <p style={styles.saveBtn} >Save</p>
          </span>
          <p style={styles.characterCount}>{text.length} characters</p>
        </div>
        <Link link={linkToOpen} text='Click here to open'/>
      </div>
    );
  }
}

Home.defaultProps = {};

Home.propTypes = {
  setTitle: PropTypes.func,
  setText: PropTypes.func,
  title: PropTypes.string,
  saveNote: PropTypes.func,
  notes: PropTypes.array,
  text: PropTypes.string,
  linkToOpen: PropTypes.string
};

export default Home;
