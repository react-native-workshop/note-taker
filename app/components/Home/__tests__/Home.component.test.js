import React from 'react';
import renderer from 'react-test-renderer';
import Home from '../Home.component';

describe('Home component', () => {
  it('renders correctly', () => {
    const component = renderer.create(<Home />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
