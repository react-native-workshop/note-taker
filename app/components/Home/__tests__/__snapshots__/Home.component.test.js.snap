// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "padding": 20,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
  >
     Please enter your note here
  </Text>
  <TextInput
    multiline={true}
    onChangeText={[Function]}
    style={
      Array [
        Object {
          "borderColor": "gray",
          "borderWidth": 1,
          "height": 100,
          "width": 200,
        },
        undefined,
      ]
    }
    value=""
  />
</View>
`;
